src/http.cc

```diff
@@ -75,10 +75,11 @@ static void copyOneHeaderFromClientsideRequestToUpstreamRequest(const HttpHeader
         HttpHeader * hdr_out, const int we_do_ranges, const Http::StateFlags &);
 
 HttpStateData::HttpStateData(FwdState *theFwdState) :
+    _peer(nullptr),
     AsyncJob("HttpStateData"),
     Client(theFwdState),
```

Comment:
We should initialize parent classes first:

```suggestion
    AsyncJob("HttpStateData"),
    Client(theFwdState),
    _peer(nullptr),
```

You can also  initialize `_peer`  in the class declaration instead so that you do not have to worry about the proper initialization order. We already initialize `upgradeHeaderOut` that way AFAICT.

Making all HttpStateData initializations consistent (e.g., moving `payloadTruncated` and `surrogateNoStore` initialization to the header file) may be declared in-scope as well, but that is optional. Your call.

Reply:
I'm moving all to the header. It allows for more clarity IMO and might enable optimizations by the compiler